src/Traits/MockClassTrait.php

```diff
@@ -31,12 +33,14 @@ public function mockClass(string $class, array $callChain): void
             $mock
                 ->expects($this->exactly($calls->count()))
                 ->method($method)
-                ->withConsecutive(...$calls->map(function ($call) {
-                    return $call['arguments'];
-                })->toArray())
-                ->willReturnOnConsecutiveCalls(...$calls->map(function ($call) {
-                    return $call['result'];
-                })->toArray());
+                ->with(
```

Comment:
as discussed, please try to use `willReturnCallback` method to assert input args correctly

Reply:
@DenTray Done